lib/archethic/shared_secrets/node_renewal.ex

```diff
@@ -114,6 +114,35 @@ defmodule ArchEthic.SharedSecrets.NodeRenewal do
     )
   end
 
+  @spec new_origin_shared_secrets_transaction(
+          origin_public_keys :: list(Crypto.key()),
+          secret_key :: binary()
+        ) :: Transaction.t()
+  def new_origin_shared_secrets_transaction(
+        origin_public_keys \\ [<<0::0, 0::0, :crypto.strong_rand_bytes(32)::binary>>],
+        secret_key \\ :crypto.strong_rand_bytes(32)
+      )
+      when is_binary(secret_key) and
+             is_list(origin_public_keys) do
+    {new_shared_origin_pub_key, new_shared_origin_priv_key} = Crypto.generate_random_keypair()
+    origin_public_key = Enum.at(origin_public_keys, 0)
```

Comment:
If you are taking a list of public keys, why are you online encrypting for the first ?